pandas/core/indexes/datetimes.py

```diff
@@ -915,6 +915,45 @@ def to_series(self, keep_tz=False):
                       index=self._shallow_copy(),
                       name=self.name)
 
+    def to_frame(self, index=True, keep_tz=False):
+        """
+        Create a DataFrame with a column containing the DatetimeIndex.
+
+        .. versionadded:: 0.21.0
+
+        Parameters
+        ----------
+        index : boolean, default True
+            Set the index of the returned DataFrame as self.
+
+        keep_tz : optional, defaults False.
+            return the data keeping the timezone.
```

Comment:
Is there a reason to have this keyword? 
I suppose the reason is for consistency with `to_series`, but that has this keyword for historical reasons (I think): it dates from before we could have tz aware data inside a series / frame. 

Given that, I am not sure it is needed to copy that here.

Reply:
@jorisvandenbossche : Given that you're not 100% sure with this logic, I'd rather stick with consistency.  We can always remove (and deprecate) the argument in one sweep.

Reply:
If we only think of deprecating it, then it makes no sense to add it IMO.  
(you can say the same with "we can always add it later if people want it" :-))